context/app/static/js/shared-styles/charts/hooks.ts

```diff
@@ -0,0 +1,138 @@
+import { MouseEvent, useMemo } from 'react';
```

Comment:
Are the changes to the charts components/hooks just typescript conversions or also a mix of meaningful changes?

Reply:
Full summary of revisions to related components:

# VerticalStackedBarChart.jsx -> VerticalStackedBarChart.tsx

- Allow setting custom X/Y scales via `getXScaleRange`/`getYScaleRange` to allow changing scales
- Styling updates for labels to make them match the latest designs
- Allow formatting Y axis tick values with the `getTickValues` prop
- Allow using log scales in general

# hooks.js -> hooks.ts

- Added hooks to create linear, log, band, and ordinal d3 scales

# types.ts
- Added util function to distinguish between tooltips with and without bar data (i.e. tooltips on axis labels vs. tooltips on data)

# utils.js -> utils.ts

- Improved type safety of `trimStringWithMiddleEllipsis` so it doesn't crash when passed a null or undefined value
- Added `defaultXScaleRange` / `defaultYScaleRange`